<launch>
  <!-- AprilTag Detection Launch File for Lane Following Integration -->
  
  <arg name="robot_name" default="blueduckie"/>
  
  <!-- AprilTag Detector Node -->
  <node name="apriltag_detector" pkg="myapriltags" type="apriltag_detector.py" output="screen">
    <param name="robot_name" value="$(arg robot_name)"/>
    <!-- AprilTag detection parameters -->
    <param name="tag_family" value="tag36h11"/>
    <param name="tag_size" value="0.065"/>  <!-- Size in meters -->
    <param name="max_hamming" value="0"/>
    <param name="detection_window_width" value="640"/>
    <param name="detection_window_height" value="480"/>
    <!-- Remap camera topic -->
    <remap from="/camera_node/image/compressed" to="/$(arg robot_name)/camera_node/image/compressed"/>
  </node>
  
  <!-- AprilTag Controller for Lane Integration -->
  <node name="apriltag_controller" pkg="myapriltags" type="apriltag_controller.py" output="screen">
    <param name="robot_name" value="$(arg robot_name)"/>
    <!-- Control parameters -->
    <param name="approach_distance" value="0.3"/>
    <param name="alignment_tolerance" value="0.1"/>
    <param name="max_approach_speed" value="0.2"/>
  </node>
  
  <!-- AprilTag Lane Integration -->
  <node name="apriltag_lane_integration" pkg="myapriltags" type="apriltag_lane_integration.py" output="screen">
    <param name="robot_name" value="$(arg robot_name)"/>
    <!-- Integration parameters -->
    <param name="localization_weight" value="0.3"/>
    <param name="lane_following_weight" value="0.7"/>
    <param name="switch_threshold" value="0.5"/>
  </node>

</launch>